### PR TITLE
Use PyTorch to shuffle the queue patches

### PR DESCRIPTION
**Pull request by fepegar**
Related to:
- #766

**Description**
The queue was using the built-in `random` module to shuffle the patches. This commit is to use PyTorch, so reproducibility can be handled more easily. To obtain fully reproducible results in a TorchIO program:

```python
import torch

seed = 20345978  # for example
torch.manual_seed(20345978)
torch.use_deterministic_algorithms(True)
```

More info can be found at #766 and the [PyTorch docs on reproducibility](https://pytorch.org/docs/stable/notes/randomness.html).

**Checklist**

- [x] I have read the [`CONTRIBUTING`](https://github.com/fepegar/torchio/blob/main/CONTRIBUTING.rst) docs and have a developer setup (especially important are `pre-commit`and `pytest`)
- [x] Non-breaking change (would not break existing functionality)
- [ ] Breaking change (would cause existing functionality to change)
- [ ] Tests added or modified to cover the changes
- [x] Integration tests passed locally by running `pytest`
- [ ] In-line docstrings updated
- [ ] Documentation updated, tested running `make html` inside the `docs/` folder
- [x] This pull request is ready to be reviewed
- [ ] If the PR is ready and there are multiple commits, I have [squashed them and force-pushed](https://www.w3docs.com/snippets/git/how-to-combine-multiple-commits-into-one-with-3-steps.html#force-pushing-commits-7)

### CODE REVIEW

**Review by dzenanz — APPROVED** [review]

**Comment by fepegar** [comment]
Thanks, @dzenanz!

**Comment by fepegar** [comment]
@all-contributors please add @dzenanz for reviewing this PR and reporting the reproducibility results.

**Comment by fepegar** [comment]
@all-contributors please add @dzenanz for review

**Comment by dzenanz** [comment]
Awesome, thank you!
